src/Purchase.java

```diff
@@ -0,0 +1,72 @@
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Purchase {
+    private final String NAME;
+    private Byn price;
+    private int number;
+
+    public Purchase() {
+        NAME = "";
+        price = new Byn();
+    }
+
+    public Purchase(String name, Byn price, int number) {
+        this.NAME = name;
```

Comment:
if you use constanse - don't use it in the constructor. Or replace it with simpe field